Voycar.Api.Web/Features/Members/Endpoints/Get/Personal/Endpoint.cs

```diff
@@ -0,0 +1,50 @@
+namespace Voycar.Api.Web.Features.Members.Endpoints.Get.Personal;
+
+using Repository;
+using Users.Repository;
+
+
+public class Endpoint : Endpoint<Request, Results<Ok<Response>, BadRequest<string>>, Mapper>
+{
+    private readonly IUsers _userRepository;
+    private readonly IMembers _memberRepository;
+
+    public Endpoint(IUsers userRepository, IMembers memberRepository)
+    {
+        this._userRepository = userRepository;
+        this._memberRepository = memberRepository;
+    }
+
+
+    public override void Configure()
+    {
+        this.Get("member/personal");
+        this.Roles("member");
+        this.Summary(summary =>
+        {
+            summary.Summary = "Get personal details of logged in member";
+            summary.Description = "Get the personal details of the member who is logged in according to the request cookie";
+            summary.Responses[200] = "If member is found";
+            summary.Responses[400] = "If the user somehow can't be found or the user is not a member.";
+        });
+    }
+
+    public override async Task HandleAsync(Request req, CancellationToken ct)
+    {
+        var user = this._userRepository.Retrieve(req.UserId);
+        if (user is null)
+        {
+            await this.SendResultAsync(TypedResults.BadRequest("User does not exist"));
+            return;
+        }
+
+        var member = this._memberRepository.Retrieve(user.MemberId.Value);
+        if (member is null)
+        {
+            await this.SendResultAsync(TypedResults.BadRequest("Member does not exist"));
+            return;
+        }
+
+        await this.SendResultAsync(TypedResults.Ok(this.Map.FromEntities(member!, user)));
```

Comment:
Der folgende Vorschlag mag jetzt etwas doppelt wirken, aber die ID kann `null` sein und wenn sie `null` ist kommt es zu Fehlermeldungen. 
![grafik](https://github.com/solid-stack-solutions/voycar-backend/assets/58894308/d5d1ca51-df29-4a91-a753-ebefa3cb4527)

Das Ausrufezeichen kann jetzt aber weg, weil wir ja `null` geprüft haben.
```suggestion
        }
        if (user.MemberId is null)
        {
            await this.SendResultAsync(TypedResults.BadRequest("Member does not exist"));
            return;
        }

        var member = this._memberRepository.Retrieve(user.MemberId.Value);
        if (member is null)
        {
            await this.SendResultAsync(TypedResults.BadRequest("Member does not exist"));
            return;
        }

        await this.SendResultAsync(TypedResults.Ok(this.Map.FromEntities(member, user)));
```

Reply:
Und das gibt dann auch eine Warning in Rider?

Reply:
Yep, ohne die Änderungen gibt es eine Warnung. Und ohne das Value kommt eine Fehlermeldung, dass die Methode `Retrieve` kein nullable Guid akzeptiert.
![grafik](https://github.com/solid-stack-solutions/voycar-backend/assets/58894308/961cce09-25b3-45ec-9145-541209ed3f30)